plugins/catalog-backend-module-ldap/src/ldap/read.ts

```diff
@@ -219,17 +221,17 @@ export async function readLdapGroups(
   groupMemberOf: Map<string, Set<string>>; // DN -> DN or UUID of groups
   groupMember: Map<string, Set<string>>; // DN -> DN or UUID of groups & users
 }> {
-  if (config.length === 0) {
+  if (groupConfig.length === 0) {
     return { groups: [], groupMemberOf: new Map(), groupMember: new Map() };
   }
   const groups: GroupEntity[] = [];
   const groupMemberOf: Map<string, Set<string>> = new Map();
   const groupMember: Map<string, Set<string>> = new Map();
 
-  const vendor = await client.getVendor();
+  const vendor = await client.getVendor(vendorConfig);
```

Comment:
Not really any use passing through this. The vendors can be left untouched. Instead, the code here in this file can do the fallback defaulting (taking the config if given, and otherwise using the vendor value)

Reply:
Would you suggest something like this then? (keeping vendors and getVendor unchanged) 
```suggestion
  let vendor: LdapVendor;

  if (vendorConfig) {
    vendor = {
      ...vendorConfig,
      decodeStringAttribute: (entry, name) => {
        const values = entry.raw[name];
        if (Array.isArray(values)) {
          return values.map(v => v.toString());
        } else if (values) {
          return [values.toString()];
        }
        return [];
      },
    };
  } else {
    vendor = await client.getVendor();
  }

```

Reply:
I mean more like

```ts
const vendorDefaults = await client.getVendor();
const vendor: LdapVendor = {
  dnAttributeName: config.getOptionalString(...) ?? vendorDefaults.dnAttributeName,
  uuidAttributeName: ...same pattern as above...,
  decodeStringAttribute: vendorDefaults.decodeStringAttribute,
};
```

Reply:
@freben done!